Assets/Scripts/HttpTopicClient.cs

```diff
@@ -86,30 +85,41 @@ public HttpTopicClient(string authToken){
         this.authToken = authToken;
     }
 
-    public void Publish(string cacheName, string topicName, byte[] message)
+    public IEnumerator Publish(string cacheName, string topicName, byte[] message)
     {
         Debug.Log("Publishing binary message");
-        UnityWebRequest request = new UnityWebRequest(
+        using var request = new UnityWebRequest(
             "https://" + endpoint + "/topics/" + cacheName + "/" + topicName,
             "POST"
         );
         request.SetRequestHeader("Content-Type", MessageContentType.Binary);
         request.uploadHandler = new UploadHandlerRaw(message);
         request.uploadHandler.contentType = MessageContentType.Binary;
         request.SetRequestHeader("Authorization", authToken);
-        request.SendWebRequest();
+        yield return request.SendWebRequest();
+        
+        if (request.result != UnityWebRequest.Result.Success)
+        {
+            Debug.LogError("Publish failed: " + request.error);
+        }
     }
 
-    public void Publish(string cacheName, string topicName, string message)
+    public IEnumerator Publish(string cacheName, string topicName, string message)
     {
+        Debug.Log("Publishing string message");
         var contentType = MessageContentType.Text;
-        var request = UnityWebRequest.Post(
+        using var request = UnityWebRequest.Post(
             "https://" + endpoint + "/topics/" + cacheName + "/" + topicName,
             message,
             contentType
         );
         request.SetRequestHeader("Authorization", authToken);
-        request.SendWebRequest();
+        yield return request.SendWebRequest();
```

Comment:
What are the semantics of `using var request` given that you are yielding `request.SendWebRequest`?

Reply:
I believe it will go out of scope and be cleaned up when the coroutine finishes. Without the `yield return` I think the request can go out of scope and be cleaned up before the request finishes. I didn't see issues when running the example, but it felt strange trusting that the async request would complete successfully after the method returned.